File: language-support/js/daml-ledger/src/model/get_transaction_by_event_id_request.ts
// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

/**
 * @interface GetTransactionByEventIdRequest
 * @memberof ledger
 */
export interface GetTransactionByEventIdRequest {
    /**
     * @member {string} eventId
     * @memberof ledger.GetTransactionByEventIdRequest
     * @instance
     */
    eventId: string
    /**
     * @member {Array<string>} requestingParties
     * @memberof ledger.GetTransactionByEventIdRequest
     * @instance
     */
    requestingParties: string[]
}

File: sdk/canton/community/ledger/ledger-api-core/src/main/scala/com/digitalasset/canton/ledger/api/messages/transaction/GetTransactionByEventIdRequest.scala
// Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

package com.digitalasset.canton.ledger.api.messages.transaction

import com.digitalasset.canton.ledger.api.domain.EventId
import com.digitalasset.daml.lf.data.Ref.Party

final case class GetTransactionByEventIdRequest(
    eventId: EventId,
    requestingParties: Set[Party],
)
